prime-router/src/main/kotlin/Report.kt

```diff
@@ -200,6 +211,40 @@ class Report {
         return Report(schema, Table.create(columns), fromThisReport("deidentify"))
     }
 
+    // takes the data in the existing report and synthesizes different data from it
+    // the goal is to allow us to take real data in, move it around and scramble it so it's
+    // not able to point back to the actual records
```

Comment:
This looks good here for now, but as this grows, let's pull it out into a separate Synthesizer.kt file, or something like that.    Then that file can include the "default" synthesis that's currently in cli/main, so that its all in one place, and its more generally available, for example, if we wanted to generate synthetic data inside the actual hub workflow while a customer is testing.

Seems like we could have a Synthesizer.synthesizeData(inputReport: Report): Report, then we could add a SynthesizerFunction.kt azure function, that could be called as part of a hub workflow, if that's a good use case.

Reply:
The problem with doing it outside of `Report` is we lose access to the `Table` abstraction we're using in `Report` and that makes the work of synthesizing the data, moving it around, etc easier. There are notes on the code asking us not to leak the abstraction. An alternative would be to subclass `Report` and then add a `toReport` function that would make it immutable again.

Reply:
Good point.  I remember this problem, and I'm afraid leakage has already occurred.   I had the same issue when I built the JurisdictionalFilters, and I just cheated and used the Table features.   

I don't like the subclass idea -doesn't seem worth it.  I support just leaving it as is for now; I think leakage is the way to go over the long term.

Reply:
I like not leaking the Table abstraction. I'm not sure we want to keep this table package in the long term.